metals/src/main/scala/scala/meta/internal/metals/codeactions/BracelessBracefulSwitchCodeAction.scala

```diff
@@ -0,0 +1,558 @@
+import scala.meta.internal.metals.CodeAction
+import scala.meta.internal.metals.MetalsEnrichments._
+import scala.meta.internal.parsing.Trees
+import scala.meta.io.AbsolutePath
+import scala.meta.pc.CancelToken
+
+import org.eclipse.lsp4j.CodeActionParams
+import org.eclipse.lsp4j.TextEdit
+import org.eclipse.{lsp4j => l}
+
+class BracelessBracefulSwitchCodeAction(
+    trees: Trees,
+    buffers: Buffers
+) extends CodeAction {
+  override def kind: String = l.CodeActionKind.RefactorRewrite
+
+  override def contribute(params: CodeActionParams, token: CancelToken)(implicit
+      ec: ExecutionContext
+  ): Future[Seq[l.CodeAction]] = Future {
+    val path = params.getTextDocument().getUri().toAbsolutePath
+    val range = params.getRange()
+    val maybeTree =
+      if (range.getStart == range.getEnd)
+        trees
+          .findLastEnclosingAt[Tree](
+            path,
+            range.getStart(),
+            applyWithSingleFunction
+          )
+      else None
+
+    val result =
+      for {
+        tree <- maybeTree
+        document <- buffers.get(path)
+      } yield {
+        tree match {
+          case _: Pkg => None.toSeq
+          case classDefn: Defn.Class =>
+            createCodeActionForTemplateHolder(
+              classDefn,
+              path,
+              document,
+              classDefn.templ,
+              "class definition"
+            ).toSeq
+          case objectDefn: Defn.Object =>
+            createCodeActionForTemplateHolder(
+              objectDefn,
+              path,
+              document,
+              objectDefn.templ,
+              "object definition"
+            ).toSeq
+          case traitDefn: Defn.Trait =>
+            createCodeActionForTemplateHolder(
+              traitDefn,
+              path,
+              document,
+              traitDefn.templ,
+              "trait definition"
+            ).toSeq
+          case enumDefn: Defn.Enum =>
+            createCodeActionForTemplateHolder(
+              enumDefn,
+              path,
+              document,
+              enumDefn.templ,
+              "enum definition"
+            ).toSeq
+          case valDefn: Defn.Val =>
+            createCodeActionForPotentialBlockHolder(
+              valDefn,
+              path,
+              document,
+              valDefn.rhs,
+              "val definition"
+            ).toSeq
+          case varDefn: Defn.Var =>
+            varDefn.rhs.flatMap {
+              createCodeActionForPotentialBlockHolder(
+                varDefn,
+                path,
+                document,
+                _,
+                "var definition"
+              )
+            }.toSeq
+          case defDefn: Defn.Def =>
+            createCodeActionForPotentialBlockHolder(
+              defDefn,
+              path,
+              document,
+              defDefn.body,
+              "def definition"
+            ).toSeq
+          case termTry: Term.Try =>
+            Seq(
```

Comment:
Could we only show one action based on the current position to avoid clutter? IF we are at a Try in the block or keyword then show the actions for Try, the same for catch and finally

Reply:
I tried that. Unfortunately, there is not a straightforward way for knowing which keyword of try, catch, or finally we are at! It just captures the whole try-catch-finally tree. You can inspect this on the ast-explorer